rust/chains/abacus-ethereum/src/retrying.rs

```diff
@@ -1,38 +1,57 @@
 use std::{fmt::Debug, str::FromStr, time::Duration};
 
 use async_trait::async_trait;
-use ethers::providers::{JsonRpcClient, ProviderError};
+use ethers::providers::{Http, JsonRpcClient, ProviderError};
 use serde::{de::DeserializeOwned, Serialize};
 use serde_json::Value;
 use thiserror::Error;
 use tokio::time::sleep;
 use tracing::{debug, instrument, warn};
 
+use crate::HttpClientError;
+
 /// An HTTP Provider with a simple naive exponential backoff built-in
 #[derive(Debug, Clone)]
 pub struct RetryingProvider<P> {
     inner: P,
     max_requests: usize,
+    base_retry_ms: u64,
 }
 
 impl<P> RetryingProvider<P> {
     /// Instantiate a RetryingProvider
-    pub fn new(inner: P, max_requests: usize) -> Self {
-        Self {
+    pub fn new(inner: P, max_requests: usize, base_retry_ms: u64) -> Self {
+        let mut zelf = Self {
             inner,
-            max_requests,
-        }
+            max_requests: 0,
+            base_retry_ms: 0,
+        };
+        zelf.set_max_requests(max_requests);
+        zelf.set_base_retry_ms(base_retry_ms);
+        zelf
     }
 
-    /// Set the max_requests (and by extension the total time a request can take)
+    /// Set the max_requests (and by extension the total time a request can take).
     pub fn set_max_requests(&mut self, max_requests: usize) {
+        assert!(max_requests >= 1);
```

Comment:
It does not work with const assert because it is not a const function. What would be the correct way to handle invalid input? Default to something arbitrary?